### ISSUE TITLE
Couldn't force OpenSeaDragon to ask png format from IIIF tileSource

### ISSUE THREAD

**Issue by Kevin-Darty** [openseadragon/openseadragon#1343]
Hi,

OpenSeaDragon is always asking for jpg files to the iiif server, even if the tileSource format is png only:
`GET https://myserver/loris/myfile.png/full/286,/0/default.jpg`

I noticed this issue for some of my files that are pngs for which asking default.jpg is returning a 500 error page:
`Server Side Error: cannot write mode RGBA as JPEG`

My JS:
```
var tileSource = 
{
  "profile": [
    {
      "supports": [
        "canonicalLinkHeader",
        "profileLinkHeader",
        "mirroring",
        "rotationArbitrary",
        "regionSquare",
        "sizeAboveFull"
      ],
      "qualities": [
        "default",
        "color",
        "gray",
        "bitonal"
      ],
      "formats": [
        "png"
      ]
    }
  ],
  "protocol": "http://iiif.io/api/image",
  "sizes": [],
  "height": 3230,
  "width": 2281,
  "@context": "http://iiif.io/api/image/2/context.json",
  "@id": "https://myserver/loris/myfile.png"
};
var jsonOpenSeaDragon = {
    ...
    tileSources: [ tileSource ],
    ...
};
OpenSeadragon(jsonOpenSeaDragon);
```

**Comment by iangilman** [comment]
Any thoughts, @jpstroop @azaroth42?

**Comment by azaroth42** [comment]
The issue is correct, the tilesource is hard coded to request .jpgs, as every server is required to support them ... but of course they don't have an alpha channel. The initialization could be extended to have a format and/or quality override to request something other than `default.jpg`

**Comment by niwizard** [comment]
Thanks for your answers.

Adding a `png` format in my case would help me with `ARGB jpgs`, indeed requesting directly `https://myserver/loris/myfile.png/full/286,/0/default.png` is working well as `png` is handling the alpha channel.

**Comment by iangilman** [comment]
Sounds good! @niwizard would you be up for writing a patch to add this functionality? @azaroth42 any suggestions on how to go about it?

**Comment by eroux** [comment]
Would something like that work?

https://github.com/openseadragon/openseadragon/compare/master...buda-base:fix-1343

it's probably worth waiting for the developments of https://github.com/IIIF/api/issues/1786 before merging it, but my question is more a general code review

**Comment by iangilman** [comment]
Looks good to me from an OSD standpoint... I don't know that much about the IIIF side.

**Comment by iangilman** [comment]
Fixed by #1625. See discussion there for strategies.
